rules/msk_module_backend_test.go

```diff
@@ -0,0 +1,135 @@
+package rules
+
+import (
+	"testing"
+
+	hcl "github.com/hashicorp/hcl/v2"
+	"github.com/stretchr/testify/require"
+	"github.com/terraform-linters/tflint-plugin-sdk/helper"
+)
+
+func Test_MskModuleBackend(t *testing.T) {
+	rule := NewMskModuleBackendRule()
+
+	tests := []struct {
+		Name     string
+		Files    map[string]string
+		Expected helper.Issues
+	}{
+		{
+			Name: "backend doesn't have the team's suffix",
+			Files: map[string]string{"backend.tf": `
+terraform {
+  backend "s3" {
+    bucket = "mybucket"
+    key    = "dummy-key"
+    region = "us-east-1"
+  }
+}`},
+			Expected: helper.Issues{
+				{
+					Rule:    rule,
+					Message: "backend key must have the team's name 'rules' as a suffix. Current value is: dummy-key",
+					Range: hcl.Range{
+						Filename: "backend.tf",
+						Start:    hcl.Pos{Line: 5, Column: 5},
+						End:      hcl.Pos{Line: 5, Column: 25},
+					},
+				},
+			},
+		},
+		{
+			Name:  "no terraform config defined",
+			Files: map[string]string{"empty.tf": ``},
+			Expected: helper.Issues{
+				{
+					Rule:    rule,
+					Message: "an s3 backend should be configured for a kafka MSK module",
+					Range:   hcl.Range{},
+				},
+			},
+		},
+		{
+			Name: "no backend defined",
+			Files: map[string]string{"env.tf": `
+terraform{
+	required_version = ">= 1.5.0"
+}`},
+			Expected: helper.Issues{
+				{
+					Rule:    rule,
+					Message: "an s3 backend should be configured for a kafka MSK module",
+					Range:   hcl.Range{},
+				},
+			},
+		},
+		{
+			Name: "backend is not s3",
+			Files: map[string]string{"backend.tf": `
+terraform {
+  backend "local" {
+  }
+}`},
+			Expected: helper.Issues{
+				{
+					Rule:    rule,
+					Message: "backend should always be s3 for a kafka MSK module",
+					Range: hcl.Range{
+						Filename: "backend.tf",
+						Start:    hcl.Pos{Line: 3, Column: 3},
+						End:      hcl.Pos{Line: 3, Column: 18},
+					},
+				},
+			},
+		},
+		{
+			Name: "backend doesn't specify properties",
+			Files: map[string]string{"backend.tf": `
+terraform {
+  backend "s3" {
+  }
+}`},
+			Expected: helper.Issues{
+				{
+					Rule:    rule,
+					Message: "the s3 backend should specify the details inside the kafka MSK module",
+					Range: hcl.Range{
+						Filename: "backend.tf",
+						Start:    hcl.Pos{Line: 3, Column: 3},
+						End:      hcl.Pos{Line: 3, Column: 15},
+					},
+				},
+			},
+		},
+		{
+			Name: "backend defined in second terraform config",
+			Files: map[string]string{
+				"env.tf": `
+terraform{
+	required_version = ">= 1.5.0"
+}`,
+				"backend.tf": `
+terraform {
+  backend "s3" {
+	bucket = "mybucket"
+	key    = "good-key-rules"
```

Comment:
the linking between the current directory names and test feels a bit confusing (why is the 'team' called 'rules'?) to me.

Since the `TestRunner` always returns the current directory for `GetOriginalwd` https://pkg.go.dev/github.com/terraform-linters/tflint-plugin-sdk/helper#Runner.GetOriginalwd maybe we can just `os.Chdir` and use some more explicit naming. Something like:

```diff
diff --git a/rules/msk_module_backend_test.go b/rules/msk_module_backend_test.go
index 9bf6ac3..67ba6bd 100644
--- a/rules/msk_module_backend_test.go
+++ b/rules/msk_module_backend_test.go
@@ -1,6 +1,8 @@
 package rules
 
 import (
+       "os"
+       "path/filepath"
        "testing"
 
        hcl "github.com/hashicorp/hcl/v2"
@@ -12,9 +14,10 @@ func Test_MskModuleBackend(t *testing.T) {
        rule := NewMskModuleBackendRule()
 
        tests := []struct {
-               Name     string
-               Files    map[string]string
-               Expected helper.Issues
+               Name       string
+               Files      map[string]string
+               Expected   helper.Issues
+               WorkingDir string
        }{
                {
                        Name: "backend doesn't have the team's suffix",
@@ -102,7 +105,8 @@ func Test_MskModuleBackend(t *testing.T) {
                        },
                },
                {
-                       Name: "backend defined in second terraform config",
+                       Name:       "backend defined in second terraform config",
+                       WorkingDir: filepath.Join("cluster-config", "my-team"),
                        Files: map[string]string{
                                "env.tf": `
 terraform{
@@ -112,7 +116,7 @@ func Test_MskModuleBackend(t *testing.T) {
 terraform {
   backend "s3" {
        bucket = "mybucket"
-       key    = "good-key-rules"
+       key    = "good-key-my-team"
        region = "us-east-1"
   }
 }`,
@@ -124,6 +128,11 @@ func Test_MskModuleBackend(t *testing.T) {
        for _, test := range tests {
                t.Run(test.Name, func(t *testing.T) {
                        runner := helper.TestRunner(t, test.Files)
+                       if test.WorkingDir != "" {
+                               workingDir := filepath.Join(t.TempDir(), test.WorkingDir)
+                               require.NoError(t, os.MkdirAll(workingDir, 0o700))
+                               require.NoError(t, os.Chdir(workingDir))
+                       }
 
                        if err := rule.Check(runner); err != nil {
                                require.NoError(t, err, "Unexpected error occurred")
```

Reply:
Indeed that is a bit misleading. Your proposal seems reasonable